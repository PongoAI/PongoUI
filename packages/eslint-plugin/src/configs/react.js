/** @type {import("eslint").Linter.Config} */
const config = {
  extends: ['airbnb-typescript', 'prettier'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: 'tsconfig.json',
  },
  plugins: ['deprecation', 'import', '@typescript-eslint', 'jest', 'jsx-a11y', 'react', 'react-hooks'],
  settings: {
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      typescript: {
        alwaysTryTypes: true,
      },
    },
    'no-unused-expressions': 'off',
  },
  env: {
    browser: true,
    es6: true,
    node: true,
  },

  ignorePatterns: ['coverage', 'dist', 'dist-storybook', 'lib', 'lib-commonjs', '**/__snapshots__', 'node_modules'],
  rules: {
    'no-unused-vars': 'error',
    'no-use-before-define': 'off',
    '@typescript-eslint/no-use-before-define': ['error'],
    'import/no-extraneous-dependencies': ['error', { devDependencies: true }],
    'import/extensions': [
      'error',
      'ignorePackages',
      {
        js: 'never',
        jsx: 'never',
        ts: 'never',
        tsx: 'never',
      },
    ],
  },
};

module.exports = config;
