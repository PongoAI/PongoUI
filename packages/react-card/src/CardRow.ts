export * from './components/CardRow/index';
