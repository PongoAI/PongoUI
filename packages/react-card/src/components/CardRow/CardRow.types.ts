import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';

export type CardRowSlots = {
  /**
   * The root of CardRow.
   */
  root: Slot<'div'>;
};

export interface CardRowProps extends ComponentProps<CardRowSlots> {}

export type CardRowState = ComponentState<CardRowSlots>;
