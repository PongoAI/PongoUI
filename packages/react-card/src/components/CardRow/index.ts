export * from './CardRow';
export * from './CardRow.types';
export * from './renderCardRow';
export * from './useCardRow';
