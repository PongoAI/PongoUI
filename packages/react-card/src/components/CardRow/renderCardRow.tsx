import * as React from 'react';
import { getSlots } from '@fluentui/react-utilities';
import type { CardRowState, CardRowSlots } from './CardRow.types';

export const renderCardRow = (state: CardRowState) => {
  const { slots, slotProps } = getSlots<CardRowSlots>(state);

  return <slots.root {...slotProps.root} />;
};
