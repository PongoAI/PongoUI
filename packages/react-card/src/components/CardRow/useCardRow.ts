import * as React from 'react';
import { getNativeElementProps } from '@fluentui/react-utilities';
import type { CardRowProps, CardRowState } from './CardRow.types';

export const useCardRow = (props: CardRowProps, ref: React.Ref<HTMLElement>): CardRowState => {
  const state: CardRowState = {
    root: getNativeElementProps('div', {
      ref,
      ...props,
    }),
    components: {
      root: 'div',
    },
  };

  return state;
};
