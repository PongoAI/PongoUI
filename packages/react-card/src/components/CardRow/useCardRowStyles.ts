import { makeStyles, mergeClasses, shorthands } from '@griffel/react';
import type { CardRowState } from './CardRow.types';
import { tokens } from '@pongo-ui/react-theme';

const useStyles = makeStyles({
  root: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    flexShrink: 0,
    ...shorthands.gap(tokens.relaxedGap),
  },
});

export const useCardRowStyles = (state: CardRowState) => {
  const styles = useStyles();

  state.root.className = mergeClasses(styles.root, state.root.className);

  return state;
};
