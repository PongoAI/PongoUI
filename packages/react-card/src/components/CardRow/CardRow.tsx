import * as React from 'react';
import { useCardRow } from './useCardRow';
import { renderCardRow } from './renderCardRow';
import { useCardRowStyles } from './useCardRowStyles';
import type { CardRowProps } from './CardRow.types';
import type { ForwardRefComponent } from '@fluentui/react-utilities';

/**
 * The CardRow component wraps items within a card and positions them.
 */
export const CardRow: ForwardRefComponent<CardRowProps> = React.forwardRef((props, ref) => {
  const state = useCardRow(props, ref);

  useCardRowStyles(state);

  return renderCardRow(state);
});
CardRow.displayName = 'CardRow';
