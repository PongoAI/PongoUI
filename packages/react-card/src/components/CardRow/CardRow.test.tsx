import * as React from 'react';
import { CardRow } from './CardRow';
import { isConformant } from '../../common/isConformant';
import { render } from '@testing-library/react';

describe('CardRow', () => {
  isConformant({
    Component: CardRow,
    displayName: 'CardRow',
  });

  it('renders a basic CardRow', () => {
    const renderedComponent = render(<CardRow>This is a test</CardRow>);

    expect(renderedComponent.baseElement).toMatchSnapshot();
  });
});
