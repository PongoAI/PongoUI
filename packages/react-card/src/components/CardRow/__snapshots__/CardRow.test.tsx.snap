// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardRow renders a basic CardRow 1`] = `
<body>
  <div>
    <div
      class="___130uk7d f22iagw f1063pyq f122n59 fi64zpg fnmxc1e fynvm4a"
    >
      This is a test
    </div>
  </div>
</body>
`;
